libraries/primitive_after.rb

```diff
@@ -0,0 +1,57 @@
+require_relative 'primitive'
+
+module Choregraphie
+  class After < Primitive
+    def initialize(&block)
+      @block = block
+    end
+
+    def register(choregraphie)
+      @choregraphie_name = choregraphie.name
+
+      choregraphie.before do
+        set_in_progress
+      end
+
+      choregraphie.cleanup do
+        if in_progress?
+          @block.call
+          set_not_in_progress
+        elsif !installed? # we are after a reinstallation, we can't know if we were in a choregraphie or not, run the block just in case
+          @block.call
+        end
+        set_installed
```

Comment:
This mean we will always have an 'installed' marker on the disk as long as a choregraphie is defined for the current node (even if it never actually entered it). Is this an acceptable side-effect?

Reply:
yes I think it is a good side effect

Reply:
actually that's necessary to put this file all the time to make sure we have it to avoid running the cleanup all the time